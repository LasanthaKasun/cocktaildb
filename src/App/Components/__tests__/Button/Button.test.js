import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Button } from "../..";

it("should render button component", () => {
  const component = renderer.create(<Button />);
  expect(component).toMatchSnapshot();
});

it("should render button with props", () => {
  render(<Button title="Back" icon="null" onClick={null} />);
  const actionBtn = screen.getByTestId("actionBtn");
  expect(actionBtn).toHaveTextContent("Back");
});
