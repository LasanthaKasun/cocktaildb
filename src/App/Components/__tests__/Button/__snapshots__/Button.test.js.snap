// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render button component 1`] = `
<button
  className="bg-blue-300 hover:bg-blue-400 text-gray-800 font-bold py-2 px-4 rounded inline-flex items-center gap-2"
  data-testid="actionBtn"
>
  <img
    alt="refresh"
    width="25"
  />
  <span />
</button>
`;
